azurerm/resource_arm_private_link_endpoint.go

```diff
@@ -0,0 +1,355 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateLinkEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateLinkEndpointCreateUpdate,
+		Read:   resourceArmPrivateLinkEndpointRead,
+		Update: resourceArmPrivateLinkEndpointCreateUpdate,
+		Delete: resourceArmPrivateLinkEndpointDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"manual_private_link_service_connection": {
```

Comment:
I think it'd be better for this to be split out into it's own resource, rather than being bundled into this one - could we make a new resource for (something like) `azurerm_private_link_endpoint_manual_connection`?

Reply:
I merged them into a single attribute block and exposed the manual one as a Boolean value.